app/uk/gov/hmrc/cdsimportsddsfrontend/services/CustomsDeclarationsService.scala

```diff
@@ -40,18 +40,24 @@ class CustomsDeclarationsService @Inject()(appConfig: AppConfig, declarationXml:
   def submit(eori: Eori, declarationViewModel: DeclarationViewModel)(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
     val declaration: Declaration = declarationViewModel.toDeclaration
     val xml = declarationXml.fromImportDeclaration(declaration)
-    submit(eori, xml)
+    val lrn = declarationViewModel.documentationAndReferences.localReferenceNumber
+    submit(eori, xml, lrn)
   }
 
-  def submit(eori: Eori, xml: Elem)(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
+  def submit(eori: Eori, xml: Elem, lrn: Option[String])(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
     httpClient.POSTString[CustomsDeclarationsResponse](
       appConfig.declarationsApi.submitEndpoint, xml.toString(), headers = headers(eori))(responseReader, hc, executionContext)
       .map { customsDeclarationsResponse: CustomsDeclarationsResponse =>
         log.info("Response from Declaration API: " + customsDeclarationsResponse);
+        lrn.map(number => saveDeclaration(eori, number))
         DeclarationServiceResponse(DeclarationXml.prettyPrintToHtml(xml), customsDeclarationsResponse.status, customsDeclarationsResponse.conversationId)
       }
   }
 
+  def saveDeclaration(eori: Eori, lrn: String)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
+    httpClient.POST(appConfig.cdsImportsddsDeclarations, lrn, Seq(CustomsHeaderNames.EoriIdentifier -> eori))
```

Comment:
Will this send the lrn as a serialised JSON object?

Reply:
no, it would be just a string value though when I checked the request body, it looked like this `AnyContentAsJson("2ab9ea27f1f84a70a68e")` where `2ab9ea27f1f84a70a68e` is urn value